src/txkube/_network.py

```diff
@@ -341,8 +371,38 @@ class _NetworkKubernetes(object):
         default=attr.Factory(lambda: Agent(namedAny("twisted.internet.reactor"))),
     )
 
+    def versioned_client(self):
+        """
+        Ask the server for its version and then create a client using a matching
+        model.
+        """
+        client = self.client()
+        d = client.version()
+        d.addCallback(self._version_to_client, client)
+        return d
+
+
+    _versions = {
+        (u"1", u"5"): v1_5_model,
+        (u"1", u"6"): v1_6_model,
+        (u"1", u"7"): v1_7_model,
```

Comment:
In https://github.com/LeastAuthority/kubetop/pull/44 I tried testing kubetop against `minikube start --kubernetes-version v1.7.0-alpha.2`.

It gave this error:

```
(kubetop) [richard@drax txkube]$ kubetop
Traceback (most recent call last):
Failure: twisted.internet.defer.FirstError: FirstError[#1, [Failure instance: Traceback: <type 'exceptions.ValueError'>: Unsupported Kubernetes version: .

```

The version.Info in this case looked like this:

```
version.Info(major=u'', gitTreeState=u'not a git tree', buildDate=u'1970-01-01T00:00:00Z', platform=u'linux/amd64', gitVersion=u'v0.0.0-master+$Format:%h$', gitCommit=u'$Format:%H$', compiler=u'gc', minor=u'', goVersion=u'go1.7.3')
```